app/code/Magento/Downloadable/Model/Link/ContentValidator.php

```diff
@@ -48,7 +48,7 @@ public function isValid(LinkInterface $link, $validateLinkContent = true, $valid
         if (!is_numeric($link->getPrice()) || $link->getPrice() < 0) {
             throw new InputException(__('Link price must have numeric positive value.'));
         }
-        if (filter_var($link->getNumberOfDownloads(), FILTER_VALIDATE_INT) === false
+        if (filter_var((int) $link->getNumberOfDownloads(), FILTER_VALIDATE_INT) === false
```

Comment:
I believe we should have a validator for this somewhere. Please replace both `filter_var` usages with it.